File: apps/appointment-api-service/src/domain/cache-keys.ts
export function appointmentHashKey(tenantId: string, appointmentId: string): string {
  return `tenant:${tenantId}:appointment:${appointmentId}`;
}

export function activeAppointmentsSetKey(tenantId: string): string {
  return `tenant:${tenantId}:appointments:active`;
}

export function technicianOccupiedKey(tenantId: string, technicianId: string): string {
  return `tenant:${tenantId}:technician:${technicianId}:occupied`;
}

export function bayOccupiedKey(tenantId: string, bayId: string): string {
  return `tenant:${tenantId}:bay:${bayId}:occupied`;
}

export function occupiedSlotHashKey(tenantId: string, appointmentId: string): string {
  return `tenant:${tenantId}:occupied_slot:${appointmentId}`;
}

File: apps/appointment-worker-service/src/Core/Domain/CacheKeys.cs
namespace AppointmentWorkerService.Core.Domain;

public static class CacheKeys
{
    public static string AppointmentHashKey(string tenantId, string appointmentId)
        => $"tenant:{tenantId}:appointment:{appointmentId}";

    public static string ActiveAppointmentsSetKey(string tenantId)
        => $"tenant:{tenantId}:appointments:active";

    public static string TechnicianOccupiedKey(string tenantId, string technicianId)
        => $"tenant:{tenantId}:technician:{technicianId}:occupied";

    public static string BayOccupiedKey(string tenantId, string bayId)
        => $"tenant:{tenantId}:bay:{bayId}:occupied";

    public static string OccupiedSlotHashKey(string tenantId, string appointmentId)
        => $"tenant:{tenantId}:occupied_slot:{appointmentId}";
}
